librz/il/il_validate.c

```diff
@@ -0,0 +1,798 @@
+// SPDX-FileCopyrightText: 2022 Florian Märkl <info@florianmaerkl.de>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_il/rz_il_validate.h>
+#include <ht_uu.h>
+
+/////////////////////////////////////////////////////////
+// ---------------------- context -----------------------
+
+/**
+ * Global (immutable) context
+ */
+struct rz_il_validate_global_context_t {
+	HtPP /*<const char *, RzILSortPure *>*/ *global_vars;
+	HtUU /*<RzILMemIndex, ut32:ut32>*/ *mems;
+	ut32 pc_len;
+}; /* RzILValidateGlobalContext */
+
+static void var_kv_free(HtPPKv *kv) {
+	free(kv->key);
+	free(kv->value);
+}
+
+static void var_kv_unown_free(HtPPKv *kv) {
+	free(kv->key);
+}
+
+/**
+ * Create a new global context for validation
+ * Vars and mems can be added manually with rz_il_validate_global_context_add_* functions.
+ */
+RZ_API RzILValidateGlobalContext *rz_il_validate_global_context_new_empty(ut32 pc_len) {
```

Comment:
`pc_len` - could it be zero? Are there any invalid values? Maybe add checks for them?